alyssa_mckee/class_assignments/hospital.py

```diff
@@ -0,0 +1,61 @@
+class Hospital(object):
+	def __init__(self, name, capacity):
+		self.beds = [None]*capacity
+		self.name = name
+		self.capacity = capacity
+	def admit(self, patient):
+		#find the first empty bed. put the patient there
+		for num in range(len(self.beds)):
+			if self.beds[num] == None:
+				self.beds[num] = patient
+				patient.bed = num
+				print("Admission complete")
+				break
```

Comment:
What's the use for having a break here? It seems the issue is more that your else statement is not inside the for loop

Reply:
The else is outside the for loop because I want to check all of the beds for patients before printing "the hospital if full" 

When a loop finishes iterating, it will run the else block following it, but if the loop is stopped prematurely with a `break`, it skips over that else block. (I assume this is because the loop's condition is still True when I break? I'll have to look into why this is) anyways, this is my intention. If the for loop completes its course and none of the beds were empty, only then print "the hospital is full". If the loop comes across an empty bed, put the patient there. there is no need to continue checking the rest of the beds so break out of the loop, which will also conveniently skip over the else block. 

Though now that I think about it, I could have used `return self` there instead of `break`, which would have made the else after the for loop unnecessary

Reply:
Yep, I agree that a return statement would make much more sense in this case.